### DISCUSSION TITLE
Difference between DFI (UTXO) and DFI (Token)

### DISCUSSION THREAD

**Discussion by izzycsy** [DeFiCh/app#846] [Guide]
<img width="773" alt="Insufficient utxoDFI" src="https://user-images.githubusercontent.com/65014479/117921817-fcb6d580-b323-11eb-98b0-5c8f7f598ed6.png">

![Wait for tx in next block](https://user-images.githubusercontent.com/65014479/117921828-ffb1c600-b323-11eb-94d0-59c552ad9f49.jpeg)

For users who face the above shown error messages, it’s likely that you've insufficient **DFI UTXO balance (min. 1 DFI)** — which is required to perform swap and all other functions. 

DFI exist in 2 forms; `DFI UTXO` and `DFI Token`. DFI UTXO is used to pay for transactions fees while rewards received from liquidity mining (for example) are in DFI Token. You can freely convert between the 2 anytime — at a 1:1 ratio. To convert DFI Token to DFI UTXO, the simplest way would be via DeFi Wallet app.   

#### Check your DFI on your app screen

<img width="762" alt="Balances" src="https://user-images.githubusercontent.com/65014479/131375381-6dc63ebb-97c7-4607-92fe-b1e917288fad.png">

#### Alternatively, you can check via the CLI

<img width="240" alt="CLI v2 5 1" src="https://user-images.githubusercontent.com/65014479/123064503-e92d7d00-d440-11eb-9f1f-c93b3127982e.png">

Firstly, check via CLI: 
 `getbalance` — DFI UTXO
`gettokenbalances` — DFI Token 

From your output (for example):

```
gettokenbalances
[
  "3.123456768@0", — DFI Token
  "XXXXXXXXXX@2",
  "XXXXXXXXXX@5"
]
```

#### To convert, send 1 DFI (Token) to yourself:
1. Copy your address (the one that you're planning to use to add liquidity / swap / send out to an external address)
2. Click `send DFI`
3. Input 1 DFI 
4. Paste the address (from step 1)

By sending to yourself, you're converting your `DFI Token` to `DFI UTXO`. `DFI UTXO` is also used for transaction fees.

5. If you're planning to add or remove liquidity / swap on DEX, try the function again
5. If you're planning to send out DFI (UTXO), try the function again with your external DeFiChain address

**Comment by izzycsy** [comment]
The following example is to further illustrate on how to convert `DFI (Token)` to `DFI (UTXO)` (_especially for users with multiple addresses_ ), check via the CLI for the breakdown of tokens: 

Let's say, I've 3 addresses which contains DFI — this represents the sum of DFI (UTXO) & DFI (Token).
```
listaccounts {} false false true
[
{
"key": "a9149...@0",
"owner": "degM...",
"amount": "1.60202512@DFI"
},
{
"key": "a9149...@0",
"owner": "dMwt...",
"amount": "0.40091919@DFI"
},
{
"key": "a914...@0",
"owner": "8JaZ...",
"amount": "0.50116422@DFI"
}
]
```

And let's say;
```
getbalance 
1.00000000 — DFI (UTXO)

gettokenbalances
[
"1.50419853@0", — DFI (Token)
"XXXXXXXXXX@2",
"XXXXXXXXXX@5"
]
```

The above adds up to a total of `2.50419853` DFI (sum of DFI UTXO & DFI Token).
At this point, what I'd do is stick to one address for all DFI. In the example, I'll take the first address `degM...`, and I want to convert everything to DFI UTXO. So,

1. Copy address (`degM...`)
2. Go to `SEND DFI`
3. Input `1.50419853` DFI or `MAX` 

>  or you can try converting 1 DFI Token to DFI UTXO first

4. Paste the address `degM...`
5. `SEND` button

Wait for the balance to reflect, might take a few blocks. 
By sending to myself, I'm converting my DFI Token to DFI UTXO on the address `degM...`. You need your DFI to be in UTXO form if you want to send it out of the DeFiChain Network.

From here onwards, I'll stick to using `degM...`, so I don't have to worry about finding which address holds the rewards I received etc., because it's all on `degM...`. It's also advisable to `backup wallet` for your own safety. 

<img width="269" alt="Menu bar   Backup Wallet" src="https://user-images.githubusercontent.com/65014479/122753019-3801d800-d2c4-11eb-98d3-7d76529139cf.png">

**Reply by marktmaxwell** [reply]
This "fix" also worked with the below pasted error message.
![image](https://user-images.githubusercontent.com/94632532/150156636-d7db3307-25de-4143-a514-5cf7645f64b0.png)

**Comment by plx2000** [comment]
when I want to exchange DFI to UTXO as described above, I get the error message that I don't have enough UTXO as transaction fees. Am I doing something wrong?

**Comment by pmjlaw** [comment]
![Screenshot 2022-09-05 at 10 55 12](https://user-images.githubusercontent.com/88074488/188558819-01617230-539a-4483-801e-0dc0ff1746bf.png)
Could you please help with this error, the solutions here unfortunately did not work.

**Reply by yhurynovich** [reply]
I'm having the same UTXO hell in v4.0.9. Unable to create UTXO without UTXO. Why the chain even allow zero balance UTXO ?
